Additional debug log censoring (EXPOSUREAPP-4451)

d4rken opened the pull request:
Adds censoring for diary locations, persons and the QRCode GUID.

Because diary data is stored asynchronously, we need to delay logging by a little bit such that the data (what to censor) is available to the censor modules.

To improve "dogfooding" for the test builds, I've changed the code to execute what happens in non-tester builds and then just overwrite the censored line again.

## Testing
* Check censoring
    * Scan a QRCode check GUID censoring
    * Create diary persons, should be censored as `Person#<Number>`
    * Create diary locations, should be censored as `Location#<Number>`
    * Export the diary, any logged export data should also be censored
* Check logging of uncaught exceptions again

d4rken committed:
Add log censoring for diary locations, persons and QR code token.

ralfgehrer commented on Corona-Warn-App/src/test/java/de/rki/coronawarnapp/bugreporting/censors/DiaryPersonCensorTest.kt:
```diff
@@ -0,0 +1,86 @@
+
+        every { CWADebug.isDeviceForTestersBuild } returns true
+        instance.checkLog(censorMe) shouldBe censorMe.copy(
+            message = "Ralf needs more coffee, but Matthias has had enough for today."
```
😄

ralfgehrer commented:
Minor: Noticed the double-
<img width="397" alt="Screenshot 2021-01-08 at 19 10 02" src="https://user-images.githubusercontent.com/56824/104051684-cd168600-51e8-11eb-9f38-f7e2c81b6e5b.png">
line on the regular build

ralfgehrer reviewed (commented):
Persons are censored but Locations are not.

Excerpt of the log:
```
Die nachfolgende Liste dient dem zuständigen Gesundheitsamt zur Kontaktnachverfolgung gem. § 25 IfSG.

09.01.2021 Person#2
09.01.2021 Person#1
09.01.2021 Berlin
)
```

harambasicluka commented:
@d4rken can this be merged? So is currently everything for production disabled?

d4rken commented:
> @d4rken can this be merged? So is currently everything for production disabled?

Need to fix the comments, but yes after that can be merged.
For production it needs to be started via menu and you can't get to the menu on production currently.

d4rken committed:
Fix divider not being hidden if bug report entry is hidden.

d4rken committed:
Fix censor modules not being applied in succession if multiple modules match.

d4rken commented:
> Minor: Noticed the double-line on the regular build

Addressed.

> Persons are censored but Locations are not.

Fixed, censor modules were not all applied if multiple modules match the same line.

ralfgehrer reviewed (dismissed):
LGTM. Tested on Pixel (Android 10)

ralfgehrer committed:
Merge branch 'release/1.11.x' into feature/4451-debug-log-censoring-part2

ralfgehrer committed:
Merge branch 'release/1.11.x' into feature/4451-debug-log-censoring-part2

chiljamgossow reviewed (commented):
Tested on Xiaomi device.
I found registration tokens in the OkHttp and SubmissionTask logging. Do we need to censor that as well?

d4rken commented:
> I found registration tokens in the OkHttp and SubmissionTask logging. Do we need to censor that as well?

Yes but was it in the debug console or the actual text file.
In tester builds we still have the normal timer debug log panted in `CWADebug`. The censoring only happens for the written file and only in `device` builds.

ralfgehrer committed:
Remove debug helper and address registration token PR issue

ralfgehrer committed:
Merge branch 'release/1.11.x' into feature/4451-debug-log-censoring-part2

BMItter commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/DebugLogger.kt:
```diff
@@ -107,8 +108,14 @@ object DebugLogger : DebugLoggerBase() {
-                        val censoredLine = bugCensors.get().mapNotNull { it.checkLog(rawLine) }.firstOrNull()
-                        appendLogLine(censoredLine ?: rawLine)
+                        launch {
+                            // Censor data sources need a moment to know what to censor
```
I don't like the delay part too much (feels hacky)
but I dont know how to improve it right now

d4rken replied:
I also don't like it, but don't have a better solution currently either :shrug:

d4rken commented:
There will be a follow up PR to checkout registration token censoring, @chiljamgossow  found an issue.
